cwms-data-api/src/main/java/cwms/cda/data/dao/watersupply/WaterSupplyUtils.java

```diff
@@ -158,4 +167,173 @@ public static WATER_USER_CONTRACT_TAB_T toWaterUserContractTs(WaterUserContract
         contractList.add(waterUserContractObjT);
         return new WATER_USER_CONTRACT_TAB_T(contractList);
     }
+
+    public static WAT_USR_CONTRACT_ACCT_TAB_T toWaterUserContractAcctTs(WaterSupplyAccounting accounting) {
+        List<WAT_USR_CONTRACT_ACCT_OBJ_T> watUsrContractAcctObjTList = new ArrayList<>();
+        for (Map.Entry<Instant, List<PumpTransfer>> entry : accounting.getPumpAccounting().entrySet()) {
+            for (PumpTransfer transfer : entry.getValue()) {
+                WAT_USR_CONTRACT_ACCT_OBJ_T watUsrContractAcctObjT = new WAT_USR_CONTRACT_ACCT_OBJ_T();
+                WATER_USER_CONTRACT_REF_T contractRef = toContractRef(accounting.getWaterUser(),
+                        accounting.getContractName());
+                watUsrContractAcctObjT.setWATER_USER_CONTRACT_REF(contractRef);
+                watUsrContractAcctObjT.setACCOUNTING_REMARKS(transfer.getComment());
+                watUsrContractAcctObjT.setPUMP_FLOW(transfer.getFlow());
+                LOOKUP_TYPE_OBJ_T transferType = toLookupTypeO(new LookupType.Builder()
+                        .withDisplayValue(transfer.getTransferTypeDisplay())
+                        .withActive(true)
+                        .withOfficeId(accounting.getWaterUser().getProjectId().getOfficeId())
+                        .build());
+                watUsrContractAcctObjT.setPHYSICAL_TRANSFER_TYPE(transferType);
+                switch (transfer.getPumpType()) {
+                    case IN:
+                        watUsrContractAcctObjT.setPUMP_LOCATION_REF(LocationUtil
```

Comment:
As an optimization, instead of repeatedly mapping the location for every date, map all locations at the top of the method before the loops

Reply:
Moved mapping to top of method